packages/cletus/src/knowledge.ts

```diff
@@ -91,6 +91,23 @@ export class KnowledgeFile extends JsonFile<Knowledge> {
     });
   }
 
+  /**
+   * Delete knowledge entries matching a predicate function
+   */
+  async deleteWhere(predicate: (entry: KnowledgeEntry) => boolean): Promise<number> {
+    let deletedCount = 0;
+    await this.save((knowledge) => {
```

Comment:
Save returns whatever is returned inside - so move delete count inside and return it and return this.save